test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,270 @@
+import { expect } from 'chai';
+
+import { MongoLogger, SeverityLevel } from '../../src/mongo_logger';
+
+describe('class MongoLogger', function () {
+  describe('#constructor', function () {
+    context('when an invalid value is passed', function () {
```

Comment:
I don't understand the purpose of this context, 1) the constructor takes many possible values, so one test is not sufficient, and 2) the tests under the `#constructor` need to not use `resolveOptions` at all, we need to be specific about exactly what is being passed into the constructor.